skore/src/skore/sklearn/_comparison/metrics_accessor.py

```diff
@@ -157,14 +164,175 @@ def report_metrics(
+
+        indicator_favorability : bool
+            Whether to keep the Favorability column.
+
+        aggregate : Aggregate
+            How to aggregate the resulting dataframe.
+        """
+
+        def add_model_name_to_index(df: pd.DataFrame, model_name: str) -> pd.DataFrame:
+            """Move the model name from the column index to the table index."""
+            df = copy.copy(df)
+
+            # Put the model name as a column
+            df["Estimator"] = model_name
+
+            # Put the model name into the index
+            if "Label / Average" in df.index.names:
+                new_index = ["Metric", "Label / Average", "Estimator"]
+            else:
+                new_index = ["Metric", "Estimator"]
+            df = df.reset_index().set_index(new_index)
+
+            # Then drop the model from the columns
+            df.columns = df.columns.droplevel(0)
+
+            return df
+
+        def melt_df(df):
+            df_reset = df.reset_index()
+
+            if "Label / Average" in df_reset.columns:
+                id_vars = ["Metric", "Label / Average", "Estimator"]
+            else:
+                id_vars = ["Metric", "Estimator"]
+
+            melted = pd.melt(
+                df_reset,
+                id_vars=id_vars,
+                var_name="Split",
+                value_name="Value",
+            )
+
+            return melted
+
+        results = individual_results.copy()
+
+        # Pop the favorability column if it exists, to:
+        # - not use it in the aggregate operation
+        # - later to only report a single column and not by split columns
+        if indicator_favorability:
+            favorability = results[0]["Favorability"]
+            for result in results:
+                result.pop("Favorability")
+        else:
+            favorability = None
+
+        dfs_model_name_in_index = [
+            add_model_name_to_index(df, estimator_name)
+            for df, estimator_name in zip(results, estimator_names)
+        ]
+
+        dfs_melted = [melt_df(df) for df in dfs_model_name_in_index]
+
+        df = pd.concat(dfs_melted, axis=0)
+
+        if aggregate:
+            if isinstance(aggregate, str):
+                aggregate = [aggregate]
+
+            if "Label / Average" in df.columns:
+                index = ["Metric", "Label / Average"]
+            else:
+                index = ["Metric"]
+
+            df = df.pivot_table(
+                index=index,
+                columns="Estimator",
+                values="Value",
+                aggfunc=aggregate,
+                sort=False,
+            )
+        else:
+            # Make sure the Split column goes [0, 1, 2, 0, 1, 2]
+            # Rather than [0, 0, 1, 1, 2, 2]
+
+            metric_order = df["Metric"].unique()
+
+            df["metric_order_index"] = df["Metric"].apply(
+                lambda x: list(metric_order).index(x)
+            )
```

Comment:
I think that we can leverage the categorical dtype from pandas:

```
            df["Metric"] = df["Metric"].astype(
                pd.CategoricalDtype(df["Metric"].unique(), ordered=True)
            )
            df["metric_order_index"] = df["Metric"].cat.codes
```

Reply:
It is weird that it is outdated but I would prefer this version. The list conversion and the index check looks not optimal.